src/MobileDevelopment/MobileDevelopment/MobileDevelopment/Models/CoordinateMH.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Text.RegularExpressions;
+using MobileDevelopment.Extensions;
+
+namespace MobileDevelopment.Models
+{
+    public class CoordinateMh
+    {
+        private uint _second;
+        private int _degree;
+        private uint _minute;
+        public readonly Direction Direction;
+        
+        public CoordinateMh()
+        {
+            Direction = Direction.N;
+            Degree = 0;
+            Minute = 0;
+            Second = 0;
+        }
+
+        public CoordinateMh(Direction direction, int degree, uint minute, uint second)
+        {
+            Direction = direction;
+            Degree = degree;
+            Minute = minute;
+            Second = second;
+        }
+
+        public CoordinateMh(float decimalDegree, Direction direction)
+        {
+            Direction = direction;
+            Degree = (int) Math.Truncate(decimalDegree);
+            Minute = (uint) Math.Truncate((decimalDegree - Degree) * 60);
+            Second = (uint) Math.Truncate(((decimalDegree - Degree) * 60 - Minute) * 60);
+        }
+
+        public CoordinateMh(string coordinate)
+        {
+            // handle direction.
+            if (coordinate.Contains("S"))
+            {
+                Direction = Direction.S;
+            }
+            else if (coordinate.Contains("W"))
+            {
+                Direction = Direction.W;
+            }
+            else if (coordinate.Contains("E"))
+            {
+                Direction = Direction.E;
+            }
+            else if (coordinate.Contains("N"))
+            {
+                Direction = Direction.N;
+            }
+            else
+            {
+                throw new ArgumentException("Incorrect format, direction not exist.");
```

Comment:
In case input validating may be you should place regex at the start of the ctr?

Reply:
![image](https://user-images.githubusercontent.com/43444946/107163309-941d2d00-69b1-11eb-950e-ed1aa657b8b6.png)

For now I don't  how better create regex, so I'll leave this code

Reply:
![image](https://user-images.githubusercontent.com/43444946/107192490-57bff000-69f6-11eb-9620-7940f05dab93.png)

Added validation for coordinate format